### PR TITLE
Add ShowPlaylistChip

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add `ShowPlaylistChip`.

Screen Shot 2022-06-30 at 14 12 21

#### WHY

In order to provide common components.

#### HOW

- Add `ShowPlaylistChip` implementation from media toolkit project;
- Add `placeholder` param;
- Add previews;
- Add tests;

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+fun ShowPlaylistChipPreviewNoName() {
+    ShowPlaylistChip(
+        artworkUri = "artworkUri",
+        name = null,
+        onClick = {},
+        placeholder = rememberVectorPainter(image = Icons.Default.FeaturedPlayList)
+    )
+}
+
+@Preview(
```
Comment (yschimke) [inline media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt:71]: I wonder if we can use `Very long name` methods names, and reuse shared previews?

File: media-ui/src/test/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipTest.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Test
+    fun givenArtwork_thenDisplaysArtwork() {
+        paparazzi.snapshot {
+            Box(modifier = Modifier.background(Color.Black), contentAlignment = Alignment.Center) {
+                ShowPlaylistChip(
+                    artworkUri = "artworkUri",
+                    name = "Playlists",
+                    onClick = {},
+                    placeholder = rememberVectorPainter(image = Icons.Default.FeaturedPlayList)
```
Comment (yschimke) [inline media-ui/src/test/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipTest.kt:55]: Can we apply some Tint? The black isn't a great demo of the feature.

**Review by yschimke — APPROVED** [review]

File: media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt (hunk 1 above)
Comment (luizgrp) [reply media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt:71]: not sure I understand the suggestion, do you mean to create and reuse this annotation?

```kotlin
@Preview(
    name = "Very long name",
    backgroundColor = 0xff000000,
    showBackground = true,
)
public annotation class VeryLongNamePreview
```

Comment (luizgrp) [reply media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt:71]: not sure I understand the suggestion, do you mean to create and reuse this annotation?

```kotlin
@Preview(
    name = "Very long name",
    backgroundColor = 0xff000000,
    showBackground = true,
)
public annotation class VeryLongNamePreview
```

File: media-ui/src/main/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChip.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import com.google.android.horologist.media.ui.ExperimentalHorologistMediaUiApi
+import com.google.android.horologist.media.ui.components.MediaArtwork
+
+/**
+ * [Chip] to show all items in the selected category.
+ */
+@ExperimentalHorologistMediaUiApi
+@Composable
+public fun ShowPlaylistChip(
+    artworkUri: String?,
```
[the remark this answers is not in the document]
Comment (luizgrp) [reply media-ui/src/main/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChip.kt:38]: sounds good, will follow up in a separate PR so it's also done for MediaArtwork

File: media-ui/src/test/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipTest.kt (hunk 2 above)
Comment (luizgrp) [reply media-ui/src/test/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipTest.kt:55]: cool, will follow up in a separate PR so it's done for MediaChip as previously requested, and also so the approach can be discussed separately

File: media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt (hunk 1 above)
Comment (yschimke) [reply media-ui/src/debug/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipPreview.kt:71]: I meant

```
@WearComposablePreview
@Composable
fun `No Artwork`() {
    ShowPlaylistChip(
```    

But that apparently doesn't work on Android projects. TIL.

**Comment by luizgrp** [comment]
merging with red build after 7 attempts to get green, will follow up in getting the commit green in main

File: media-ui/src/test/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipTest.kt (hunk 2 above)
Comment (luizgrp) [reply media-ui/src/test/java/com/google/android/horologist/media/ui/components/actions/ShowPlaylistChipTest.kt:55]: https://github.com/google/horologist/pull/302

**Pull request merged**